lib/assetmanager.js

```diff
@@ -168,32 +169,34 @@
      * @param {object} compComponents a lookup of components by component Id
      */
     AssetManager.prototype._addComponentsForComp = function (comp, doc, compComponents) {
-        this._componentManager.findAllComponents(comp).forEach(function (result) {
-            if (result.component) {
-                if (!result.component.default) {
-                    try {
-                        result.component.document = doc;
-                        result.component.comp = comp;
-                        result.component.assetPath = result.component.file;
-                        this._componentManager.addLayerCompComponent(result.component);
-                        compComponents.push(result.component);
-                    } catch (err) {
-                        this._errorManager.addError(comp, err.message, "Layer Comp");
+        return this._componentManager.findAllComponents(comp).then(function (components) {
+            components.forEach(function (result) {
+                if (result.component) {
+                    if (!result.component.default) {
+                        try {
+                            result.component.document = doc;
+                            result.component.comp = comp;
+                            result.component.assetPath = result.component.file;
+                            this._componentManager.addLayerCompComponent(result.component);
+                            compComponents.push(result.component);
+                        } catch (err) {
+                            this._errorManager.addError(comp, err.message, "Layer Comp");
+                        }
+                    } else {
+                        this._errorManager.addError(comp,
+                            "Default spec in layer comp names are unsupported.", "Layer Comp");
                     }
                 } else {
-                    this._errorManager.addError(comp,
-                        "Default spec in layer comp names are unsupported.", "Layer Comp");
-                }
-            } else {
-                if (result.errors) {
-                    result.errors.forEach(function (errMsg) {
-                        this._errorManager.addError(comp, errMsg, "Layer Comp");
-                    }.bind(this));
-                    
-                } else {
-                    console.warn("result.component not found, instead " + JSON.stringify(result));
+                    if (result.errors) {
+                        result.errors.forEach(function (errMsg) {
+                            this._errorManager.addError(comp, errMsg, "Layer Comp");
+                        }.bind(this));
+
+                    } else {
+                        console.warn("result.component not found, instead " + JSON.stringify(result));
+                    }
                 }
-            }
+            }.bind(this));
```

Comment:
This is where I wish github diff had an option to ignore leading whitespace on lines the same way Araxis does. The nutshell of this is 
```
return this._componentManager.findAllComponents(comp).then(function (components) {
            components.forEach(function (result) {
``` 
at the top, the rest is just indentation changes